### PR TITLE
Allow end users to pass in context object

### LINKED ISSUE #15: Can we pass a production flag to the puppy lib?
I'm trying to conditionally change html based off of whether or it is a production build and this would be awesome!

I think all we need to do is pass the add `isProduction` to the default options and add it to the site object in puppy.transform?

### PR DESCRIPTION
**Pull request by beahuang**
Fixes #15

### CODE REVIEW

File: lib/puppy.js [hunk 1]
```diff
@@ -20,7 +21,11 @@ async function puppy(opts = defaultOptions) {
   return new Promise((resolve, reject) => {
     // Source files from vinyl file system.
     const src = vfs.src(opts.pages);
-    const transform = puppy.transform({ ...context, publicPath: opts.publicPath });
+    const transform = puppy.transform({
+      ...context,
+      publicPath: opts.publicPath,
+      isProduction: opts.isProduction,
```
Comment (mgburns) [inline lib/puppy.js]: As discussed in Slack `isProduction` doesn't mean anything to this library, just to end users. Instead lets just accept a `context` object here and let end users decide what they want to pass in.

**Review by mgburns — CHANGES_REQUESTED** [review]
@beahuang Thanks for the contribution -- works as advertised! One change requested inline, let me know if you have any questions.

**Comment by beahuang** [comment]
@mgburns ready for another look!

**Review by mgburns — APPROVED** [review]
🔥 🔥 🔥

**Pull request merged**